### ISSUE TITLE
no-unexpected-multiline and template literal with typescript

### ISSUE THREAD

**Issue by quolpr** [eslint/eslint#11650]
**Tell us about your environment**

* eslint 6.9.0
* node v11.6.0
* npm 6.9.0
* typescript 3.3.4

**What parser (default, Babel-ESLint, etc.) are you using?**

Not sure. I use CRA 3 without ejecting.

**Please show your full configuration:**

<details>
<summary>Configuration</summary>

```js
{
  "extends": "react-app"
}
```

</details>

**What did you do? Please include the actual source code causing the issue, as well as the command that you used to run ESLint.**

```js
function myTag<T>(...args): any {
  return undefined;
}

export const test = myTag<{
  field: boolean;
}>`
  test;
`;
```

The command:
```bash
eslint --ext js,ts,ts,tsx src
```

**What did you expect to happen?**

No eslint warnings will be raised

**What actually happened? Please include the actual, raw output from ESLint.**

```
./src/test.ts
  Line 5:  Unexpected newline between template tag and template literal  no-unexpected-multiline
```

But with

```
function myTag<T>(...args): any {
  return undefined;
}

export const test = myTag<{ field: boolean }>`
  test;
`;
```

No warnings are showing. 

It especially frustrates when you are using styled-components and component have a lot of props, that will not fit the maximum line length.

**Comment by hardfist** [comment]
same issue, prettier will automatic format 
```
export const test = myTag<{ field: boolean }>`
  test;
`;
```
to 
```
export const test = myTag<{
  field: boolean;
}>`
  test;
`;
```
which will cause eslint error, which is verify frustrating

**Comment by kaicataldo** [comment]
Sorry this is so late - looks like this slipped through the cracks. Does this issue occur with the default parser? If not, this should probably be reported in the [`typescript-eslint`](https://github.com/typescript-eslint/typescript-eslint) repository.

@hardfist Prettier doesn't conform to ESLint's configuration, so if you're using Prettier for formatting you should turn any ESLint formatting rules off.

**Comment by mdjermanovic** [comment]
I believe this issue can be closed since it was fixed by #11698 (released in v6.7.2)

**Comment by aladdin-add** [comment]
Closing because there hasn't been activity for a few days. If it still occurs, please file a new issue. thanks!
